client/views/room/contextualBar/Call/Jitsi/CallJitsiWithData.js

```diff
@@ -155,41 +155,31 @@ const CallJitsiWithData = ({ rid }) => {
 	});
 
 	useEffect(() => {
-		let shouldDispose = false;
-
-		async function fetchData() {
-			if (!accepted || !jitsi) {
-				return;
-			}
-
-			const clear = () => {
-				jitsi.off('HEARTBEAT', testAndHandleTimeout);
-				jitsi.dispose();
-			};
+		if (!accepted || !jitsi) {
+			return;
+		}
 
-			try {
-				if (jitsi.needsStart) {
-					jitsi.start(ref.current);
-					await updateTimeout(rid, true);
-				} else {
-					await updateTimeout(rid, false);
-				}
-			} catch (error) {
-				dispatchToastMessage({ type: 'error', message: t(error.reason) });
-				clear();
-				handleClose();
+		const clear = () => {
+			jitsi.off('HEARTBEAT', testAndHandleTimeout);
+			jitsi.dispose();
+		};
+		const none = () => {};
+
+		try {
+			if (jitsi.needsStart) {
+				jitsi.start(ref.current);
+				updateTimeout(rid, true);
+			} else {
+				updateTimeout(rid, false);
 			}
-
-			jitsi.on('HEARTBEAT', testAndHandleTimeout);
-
-			shouldDispose = jitsi.openNewWindow;
+		} catch (error) {
+			dispatchToastMessage({ type: 'error', message: t(error.reason) });
+			clear();
+			handleClose();
 		}
+		jitsi.on('HEARTBEAT', testAndHandleTimeout);
 
-		fetchData().then(() => {
-			if (shouldDispose) {
-				jitsi.dispose();
-			}
-		});
+		return jitsi.openNewWindow ? none : clear;
```

Comment:
```suggestion
		if (!jitsi.openNewWindow) {
			return clear;
		}
```
Maybe we could change the return statement so that we could delete this `none` function, wdyt?

Reply:
I Agree. But I think its a good practice to always return a cleanup function and handle all checks inside that function.